docs/geometry.rst

```diff
@@ -173,3 +173,31 @@ specified, rounding precision will be disabled showing full precision.
 Format types ``'x'`` and ``'X'`` show a hex-encoded string representation of
 WKB or Well-Known Binary, with the case of the output matched the
 case of the format type character.
+
+.. _canonical-form:
+
+Canonical form
+--------------
+When operations are applied on geometries the result is returned according to some
+conventions.
+
+In most cases, geometries will be returned in "mild" canonical form. There is no goal
+to keep this form stable, so it is expected to change in future versions of GEOS:
+
+- the coordinates of exterior rings follow a clockwise orientation and interior
+  rings have a counter-clockwise orientation. This is the opposite of the OGC
+  specifications because the choice was made before this was included in the standard.
+- the starting point of rings can be changed, but it is undefined
+- the order of geometry types in a collection can be changed, but the order is
+  undefined
+
+When :func:`~shapely.normalize` is used, the "strict" canonical form is applied. This
+type of normalization is meant to be stable, so changes to it will be avoided if
+possible:
+
+- the coordinates of exterior rings follow a clockwise orientation and interior
+  rings have a counter-clockwise orientation
+- the starting point of rings is lower left
+- elements in collections are ordered by geometry type
```

Comment:
And then in you have multiple ones of the same type? Or does that get "normalized" in a Multi type?

Reply:
Indeed the same from the same type. It seems that those are ordered from top to bottom and from right to left.

``` py
import shapely

coll = shapely.GeometryCollection(
    [
        shapely.box(0, 0, 10, 10),
        shapely.Point(50, 50),
        shapely.MultiPoint([(50, 50), (55, 55)]),
        shapely.LineString([(60, 60), (70, 70)]),
        shapely.box(-20, -20, -30, -30),
        shapely.box(20, 20, 30, 30),
        shapely.box(-30, 20, -20, 30),
        shapely.MultiLineString([[(60, 60), (70, 70)]]),
        shapely.box(20, -20, 30, -30),
        shapely.MultiLineString([[(60, 60), (70, 70)], [(80, 80), (90, 90)]]),
    ]
)
print(f"{str(coll)=}")
print(f"{str(coll.normalize())=}")

coll2 = shapely.GeometryCollection(
    [
        shapely.box(-30, 20, -20, 30),
        shapely.box(20, -20, 30, -30),
        shapely.box(20, 20, 30, 30),
        shapely.LineString([(60, 60), (70, 70)]),
        shapely.Point(50, 50),
        shapely.MultiPoint([(50, 50), (55, 55)]),
        shapely.box(0, 0, 10, 10),
        shapely.box(-20, -20, -30, -30),
        shapely.MultiLineString([[(60, 60), (70, 70)]]),
        shapely.MultiLineString([[(60, 60), (70, 70)], [(80, 80), (90, 90)]]),
    ]
)
print(f"{str(coll2)=}")
print(f"{str(coll2.normalize())=}")
```
Result:
```
str(coll)='GEOMETRYCOLLECTION (POLYGON ((10 0, 10 10, 0 10, 0 0, 10 0)), POINT (50 50), MULTIPOINT ((50 50), (55 55)), LINESTRING (60 60, 70 70), POLYGON ((-30 -20, -30 -30, -20 -30, -20 -20, -30 -20)), POLYGON ((30 20, 30 30, 20 30, 20 20, 30 20)), POLYGON ((-20 20, -20 30, -30 30, -30 20, -20 20)), MULTILINESTRING ((60 60, 70 70)), POLYGON ((30 -20, 30 -30, 20 -30, 20 -20, 30 -20)), MULTILINESTRING ((60 60, 70 70), (80 80, 90 90)))'
str(coll.normalize())='GEOMETRYCOLLECTION (POLYGON ((20 20, 20 30, 30 30, 30 20, 20 20)), POLYGON ((20 -30, 20 -20, 30 -20, 30 -30, 20 -30)), POLYGON ((0 0, 0 10, 10 10, 10 0, 0 0)), POLYGON ((-30 20, -30 30, -20 30, -20 20, -30 20)), POLYGON ((-30 -30, -30 -20, -20 -20, -20 -30, -30 -30)), MULTILINESTRING ((80 80, 90 90), (60 60, 70 70)), MULTILINESTRING ((60 60, 70 70)), LINESTRING (60 60, 70 70), MULTIPOINT ((55 55), (50 50)), POINT (50 50))'
str(coll2)='GEOMETRYCOLLECTION (POLYGON ((-20 20, -20 30, -30 30, -30 20, -20 20)), POLYGON ((30 -20, 30 -30, 20 -30, 20 -20, 30 -20)), POLYGON ((30 20, 30 30, 20 30, 20 20, 30 20)), LINESTRING (60 60, 70 70), POINT (50 50), MULTIPOINT ((50 50), (55 55)), POLYGON ((10 0, 10 10, 0 10, 0 0, 10 0)), POLYGON ((-30 -20, -30 -30, -20 -30, -20 -20, -30 -20)), MULTILINESTRING ((60 60, 70 70)), MULTILINESTRING ((60 60, 70 70), (80 80, 90 90)))'
str(coll2.normalize())='GEOMETRYCOLLECTION (POLYGON ((20 20, 20 30, 30 30, 30 20, 20 20)), POLYGON ((20 -30, 20 -20, 30 -20, 30 -30, 20 -30)), POLYGON ((0 0, 0 10, 10 10, 10 0, 0 0)), POLYGON ((-30 20, -30 30, -20 30, -20 20, -30 20)), POLYGON ((-30 -30, -30 -20, -20 -20, -20 -30, -30 -30)), MULTILINESTRING ((80 80, 90 90), (60 60, 70 70)), MULTILINESTRING ((60 60, 70 70)), LINESTRING (60 60, 70 70), MULTIPOINT ((55 55), (50 50)), POINT (50 50))'
```